nemo_text_processing/inverse_text_normalization/he/verbalizers/date.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pynini
+from pynini.lib import pynutil
+
+from nemo_text_processing.inverse_text_normalization.he.graph_utils import (
+    NEMO_NOT_QUOTE,
+    GraphFst,
+    delete_space,
+    delete_zero_or_one_space,
+    insert_space,
+)
+
+
+class DateFst(GraphFst):
+    """
+    Finite state transducer for verbalizing date,
+        e.g. { day_prefix: "ה" day: "1" month_prefix: "ב" month: "6" year: "2012" } -> ה-1.6.2012
+    """
+
+    def __init__(self):
+        super().__init__(name="date", kind="verbalize")
+
+        day_prefix = (
+            pynutil.delete("day_prefix:")
```

Comment:
will need to change due to sparrowhawk compatibility

Reply:
what is the required change? I am not sure I understand, can you please explain?

Reply:
`day_prefix` isn't a valid property you can pass with the `date` class tag. It will work for the Python codebase, but sparrowhawk deployment will be incompatible. this will be a massive thoroughput limitation.

Reply:
sorry ^ is from me from my other git.

Reply:
We have to use prefixes in order to capture dates correctly.
What is your suggestion to fix this? Is the problem the variable name?
Why can't we add this prefix to the date graph?
I am not familiar enough with sparrowhawk to understand this comment. Please advise

Reply:
Sorry, let me explain more clearly:

1. For Sparrowhawk, class table and property tags, in this case `day_prefix`, are a limited set. You can only use class labels and property tags defined in Sparrowhawk.

See here for permitted class tags: https://github.com/google/sparrowhawk/blob/a0503e26a433fbd3a9ff81ba7a08819e4a3bb668/src/lib/semiotic_classes.pb.cc#L24-L59

See here for permitted property tags for date:

https://github.com/google/sparrowhawk/blob/a0503e26a433fbd3a9ff81ba7a08819e4a3bb668/src/lib/semiotic_classes.pb.cc#L205-L216

2. While the labels are closed, what is allowed within a given label is not closed. So you can use any arbitrary property you want. So you can replace `day_prefix` with `style` or `morphosyntactic_feature` in any place you use `day_prefix`. It just needs to be regular and managed by the verbalizer. My personal choice would just be to use `morphosyntactic_feature` since that's the general purpose for that tag. If there's multiple prefixes you're trying to manage, you can just use a trivial delimiter. (e.g. `morphosyntactic_feature: DAY_PREFIX/OTHER_PREFIX`)

3. Because Sparrowhawk is archived, this cannot be changed upstream without a fork. So it's a hard restriction (barring you want to create a PR for an archived repo....)

4. This restriction is only for Sparrowhawk. The NeMo Text Processing library proper allows trivial tags. So if you believe the change is outside the scope of the PR, you can let me know and I'll create an issue for Sparrowhawk integration with my team.